packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -376,5 +422,33 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function hasVoidReturnType(
```

Comment:
What do you think about renaming this function? maybe `hasValidReturnType`?

Because, this function doesn't actually check whether the given function's return type is specified void or not.

Reply:
i agree. that make sense!